sstables/mp_row_consumer.hh

```diff
@@ -974,6 +978,7 @@ class mp_row_consumer_m : public consumer_m {
                 return proceed::no;
             }
             if (_mf_filter->is_current_range_changed()) {
+                _reader->on_clustering_range_change();
```

Comment:
I think we're missing `on_clustering_range_change()` calls in some cases. E.g. when slicing a partition without a static row we want to immediately skip without reading the front of the partition. Currently, advance_context() will be called before reading from the data file, which will skip.